maintenance/wikia/WikiaMaps/removeImapTags.php

```diff
@@ -0,0 +1,258 @@
+	static protected $verbose = false;
+	static protected $test = false;
+
+	/**
+	 * @var WikiaMaps
+	 */
+	private $maps;
+
+	/**
+	 * @var Array
+	 */
+	private $mapsClientConfig;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Removes from articles <imap /> tags for given tiles' set id.";
+		$this->addOption( 'test', 'Test mode; make no changes', $required = false, $withArg = false, 't' );
+		$this->addOption( 'verbose', 'Show extra debugging output', $required = false, $withArg = false, 'v' );
+		$this->addOption( 'maps-db-host', 'WikiaMaps DB host', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-user', 'WikiaMaps DB user', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-pass', 'WikiaMaps DB pass', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-name', 'WikiaMaps DB name', $required = true, $withArg = true );
+		$this->addOption(
+			'tiles-set-id',
+			'Specify a tiles\' set id which leads to a group of maps which <imap /> tags should get removed',
+			$required = true,
+			$withArg = true,
+			'tsi'
+		);
+	}
+
+	static public function isVerbose() {
+		return self::$verbose;
+	}
+
+	static public function isTest() {
+		return self::$test;
+	}
+
+	/**
+	 * Print the message if verbose is enabled
+	 *
+	 * @param $msg - The message text to echo to STDOUT
+	 */
+	static public function debug( $msg ) {
+		if ( self::isVerbose() ) {
+			echo $msg . PHP_EOL;
+		}
+	}
+
+	static public function isValidInteger( $int ) {
+		return intval( $int ) > 0;
+	}
+
+	static public function isValidCityId( $cityId ) {
+		return self::isValidInteger( $cityId );
+	}
+
+	public function getMapsIdsUsingTileset( $cityId, $tilesSetId ) {
+		$dbw = new DatabaseMysqli(
+			$this->getOption('maps-db-host'),
+			$this->getOption('maps-db-user'),
+			$this->getOption('maps-db-pass'),
+			$this->getOption('maps-db-name')
+		);
+
+		return $dbw->selectFieldValues( 'map', 'id', [
+				'city_id' => $cityId,
+				'tile_set_id' => $tilesSetId
+		] );
+	}
+
+	public function getArticlesIdsUsingImap( $cityId ) {
+		global $wgStatsDB;
+
+		$dbw = wfGetDB(DB_SLAVE, array(), $wgStatsDB);
+		return $dbw->selectFieldValues( 'city_used_tags', 'ct_page_id', [
+			'ct_wikia_id' => $cityId,
+			'ct_kind' => 'imap'
+		] );
+	}
+
+	public function hasImapTag( $articleId, &$foundTags, &$foundTagsMapIds ) {
+		$article = Article::newFromID( $articleId );
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$results = null;
+			$articleContent = $article->getContent();
+			$imapSearchRegexp = "/<imap.*map\-id\=['\"](\d{1,})['\"].*(<\/imap|\/)>/";
+
+			$noOfFoundTags = preg_match_all($imapSearchRegexp, $articleContent, $results);
+
+			if( $noOfFoundTags === 0 ) {
+				return false;
+			}
+
+			$foundTags = $results[0];
+			$foundTagsMapIds = $results[1];
+
+			return true;
+		} else {
+			echo sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL;
```

Comment:
Is it worth logging? Also, for error output use MediaWiki's `$this->error()`, see https://github.com/Wikia/app/blob/dev/maintenance/Maintenance.php#L344